### PR TITLE
feat: new lacework_alert_channel_splunk resource

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Add new resource lacework_alert_channel_splunk

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

File: website/docs/r/alert_channel_splunk.html.markdown [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# lacework\_alert\_channel\_splunk
+
+You can use this resource enabled Lacework to forward alerts to Splunk using an HTTP Event Collector.
+
+To find more information see the [Lacework support documentation](https://support.lacework.com/hc/en-us/articles/360007889274-Splunk).
+
+## Example Usage
+
+```hcl
+resource "lacework_alert_channel_" "ops_critical" {
```
Comment (afiune) [inline website/docs/r/alert_channel_splunk.html.markdown:18]: resource name is missing

File: website/docs/r/alert_channel_splunk.html.markdown [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+subcategory: "Alert Channels"
+layout: "lacework"
+page_title: "Lacework: lacework_alert_channel_splunk"
+description: |-
+  Create and manage Splunk Alert Channel integrations
+---
+
+# lacework\_alert\_channel\_splunk
+
+You can use this resource enabled Lacework to forward alerts to Splunk using an HTTP Event Collector.
```
Comment (afiune) [inline website/docs/r/alert_channel_splunk.html.markdown:11]: this sentence sounds odd, is it missing a preposition? 🤔
```suggestion
You can use this resource to enable Lacework to forward alerts to Splunk using an HTTP Event Collector.
```

File: website/docs/r/alert_channel_splunk.html.markdown [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The Alert Channel integration name.
+* `channel` - (Optional) The Splunk channel name
```
Comment (afiune) [inline website/docs/r/alert_channel_splunk.html.markdown:35]: could we move this optional argument below to group all required arguments at the top?

File: website/docs/r/alert_channel_splunk.html.markdown [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+* `hec_token` - (Required) The token you generate when you create a new HEC input.
+* `host` - (Required) The hostname of the client from which you're sending data.
+* `port` - (Required) The destination port for forwarding events [80 or 443].
+* `ssl` - (Optional) Enable or Disable SSL.
+
```
Comment (afiune) [inline website/docs/r/alert_channel_splunk.html.markdown:40]:
```suggestion
```

File: website/docs/r/alert_channel_splunk.html.markdown [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The Alert Channel integration name.
+* `channel` - (Optional) The Splunk channel name
+* `hec_token` - (Required) The token you generate when you create a new HEC input.
+* `host` - (Required) The hostname of the client from which you're sending data.
+* `port` - (Required) The destination port for forwarding events [80 or 443].
```
Comment (afiune) [inline website/docs/r/alert_channel_splunk.html.markdown:38]: I am not 100% sure that the port could/should be only 80 or 443. That sounds like what SSL option would do 🤔-- I will verify this value internally, I have a feeling that the documentations on our site are wrong.

File: website/docs/r/alert_channel_splunk.html.markdown (hunk 3 above)
Comment (afiune) [reply website/docs/r/alert_channel_splunk.html.markdown:35]: Also, this bullet is missing a `.` at the end

File: lacework/resource_lacework_alert_channel_splunk_test.go [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+		t.Fatalf("%s must be set for acceptance tests", testAccAlertChannelSplunkIndex)
+	}
+}
+
+func testAccAlertChannelSplunkConfig(enabled bool) string {
+	return fmt.Sprintf(`
+resource "%s" "%s" {
+    name = "integration test"
+    enabled = %t
+	channel = "%s"
```
Comment (afiune) [inline lacework/resource_lacework_alert_channel_splunk_test.go:151]: indentation is off for these fields

File: lacework/resource_lacework_alert_channel_splunk.go [hunk 7]
```diff
@@ -0,0 +1,267 @@
+package lacework
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
```
Comment (afiune) [inline lacework/resource_lacework_alert_channel_splunk.go:7]: nitpick: separate internal and external imports
```suggestion
	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"

```

File: lacework/resource_lacework_alert_channel_splunk.go [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"host": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"port": {
+				Type:     schema.TypeInt,
+				Required: true,
```
Comment (afiune) [inline lacework/resource_lacework_alert_channel_splunk.go:50]: We might need to add a check to the port, but let me first research if the port is only 80 or 443, or maybe it is a range 🤔

Comment (afiune) [reply lacework/resource_lacework_alert_channel_splunk.go:50]: The schema says:
```
            "PORT": {
              "maximum": 65536,
              "minimum": 0,
              "title": "Port",
              "type": "number"
            },
```

File: lacework/resource_lacework_alert_channel_splunk.go [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+			},
+			"ssl": {
+				Type:     schema.TypeBool,
+				Optional: true,
```
Comment (afiune) [inline lacework/resource_lacework_alert_channel_splunk.go:54]: If SSL is optional, could we be explicit on the default:
```suggestion
				Optional: true,
				Default:  false,
```

**Review by afiune — COMMENTED** [review]
A few comments. looking good!

File: website/docs/r/alert_channel_splunk.html.markdown (hunk 5 above)
Comment (dmurray-lacework) [reply website/docs/r/alert_channel_splunk.html.markdown:38]: I'll remove for now.

File: website/docs/r/alert_channel_splunk.html.markdown (hunk 3 above)
Comment (dmurray-lacework) [reply website/docs/r/alert_channel_splunk.html.markdown:35]: Fixed

File: website/docs/r/alert_channel_splunk.html.markdown (hunk 2 above)
Comment (dmurray-lacework) [reply website/docs/r/alert_channel_splunk.html.markdown:11]: Fixed

File: website/docs/r/alert_channel_splunk.html.markdown (hunk 1 above)
Comment (dmurray-lacework) [reply website/docs/r/alert_channel_splunk.html.markdown:18]: Fixed

File: lacework/resource_lacework_alert_channel_splunk_test.go (hunk 6 above)
Comment (dmurray-lacework) [reply lacework/resource_lacework_alert_channel_splunk_test.go:151]: Fixed

File: website/docs/r/alert_channel_splunk.html.markdown [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+You can use this resource to enable Lacework to forward alerts to Splunk using an HTTP Event Collector.
+
+To find more information see the [Lacework support documentation](https://support.lacework.com/hc/en-us/articles/360007889274-Splunk).
+
+## Example Usage
+
+```hcl
+resource "lacework_alert_channel_splunk" "ops_critical" {
+  name      = "OPS Critical Alerts"
+  hec_token = "AA111111-11AA-1AA1-11AA-11111AA1111A"
```
Comment (afiune) [inline website/docs/r/alert_channel_splunk.html.markdown:20]:
```suggestion
  hec_token = "BA696D5E-CA2F-4347-97CB-3C89F834816F"
```

File: examples/resource_lacework_alert_channel_splunk/main.tf [hunk 11]
```diff
@@ -0,0 +1,12 @@
+provider "lacework" {}
+
+resource "lacework_alert_channel_splunk" "example" {
+  name      = "Splunk Channel Alert Example"
+  hec_token = "AA111111-11AA-1AA1-11AA-11111AA1111A"
```
Comment (afiune) [inline examples/resource_lacework_alert_channel_splunk/main.tf:5]: I found this token in the Splunk docs: https://docs.splunk.com/Documentation/Splunk/8.1.1/Data/UsetheHTTPEventCollector#JSON_request_and_response

```suggestion
  hec_token = "BA696D5E-CA2F-4347-97CB-3C89F834816F"
```

**Review by afiune — APPROVED** [review]
tenor-113382092

**Pull request merged**